src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicListener.cs

```diff
@@ -25,14 +26,15 @@ internal sealed class MsQuicListener : QuicListenerProvider, IDisposable
 
         private readonly IPEndPoint _listenEndPoint;
 
-        private sealed class State
+        internal sealed class State
         {
             // set immediately in ctor, but we need a GCHandle to State in order to create the handle.
             public SafeMsQuicListenerHandle Handle = null!;
             public string TraceId = null!; // set in ctor.
 
             public readonly SafeMsQuicConfigurationHandle? ConnectionConfiguration;
             public readonly Channel<MsQuicConnection> AcceptConnectionQueue;
+            public readonly ConcurrentDictionary<IntPtr, MsQuicConnection> PendingConnections;
```

Comment:
We're not cleaning the dictionary in case the listener gets disposed. I see that we're not cleaning `AcceptConnectionQueue` either, so I guess it's not an issue. I'm just making sure it's intentional.

Reply:
I was going back & forth on this one. With Connection & Stream we have `FlushAcceptQueue` so when the Connection is disposed, we would actively nuke all the stream.

I think for Listener everything would decompose natural since we are not playing tricks and we would release the `_stateHandle`. We could certainly speed it up by cleaning both. But I'm not sure if that matters as much as unlike connection, I would expect Listener would last long so this would be rare operation 
I think it would be easy to add. adding @stephentoub and @geoffkizer in case they have suggestions.

Reply:
With Connection & Stream we had to make sure that we release Connection's SafeHandle only after all Stream handles are released. I don't think Connection has the same relation to the Listener, so the order of finalization doesn't matter...